lib/core/util.c

```diff
@@ -416,12 +417,45 @@ h2o_iovec_vector_t h2o_extract_push_path_from_link_header(h2o_mem_pool_t *pool,
                 preload++;
             } else if (h2o_lcstris(token, token_len, H2O_STRLIT("nopush"))) {
                 nopush++;
+            } else if (h2o_lcstris(token, token_len, H2O_STRLIT("x-http2-push-only"))) {
+                push_only++;
             }
         }
+
         if (!nopush && preload)
             push_one_path(pool, &paths_to_push, &url, base_path, input_scheme, input_authority, base_scheme, base_authority);
     } while (token != NULL);
```

Comment:
How about doing the parsing and the rebuilding of filtered_value in one path?

I suggest this because it wouldn't be too difficult to retain the start and end of each link element, and use it to build filtered value at the end of this do-while loop.

The start of the element can be calculated as `url.base - 1` (or you might want to retain url with the surrounding brackets until passing the value to `push_one_path`).

The end of the element can be calculated as `token != NULL ? token : value + value_len`.